### PR TITLE
docs(core): docs for project.json files + associated generator

### PR DESCRIPTION
**Pull request by AgentEnder**
## Current Behavior
project.json files are not documented in configuration

## Expected Behavior
project.json files should be documented

### CODE REVIEW

File: docs/node/guides/configuration.md [hunk 1]
```diff
@@ -302,6 +304,33 @@ The following command will generate a new library: `nx g @nrwl/node:lib mylib`.
 
 When the `version` of `workspace.json` is set to 2, `targets`, `generators` and `executor` properties are used instead of the version 1 properties `architect`, `schematics` and `builder`.
 
+## project.json
+
+Project configurations can also be independent files, referenced by `angular.json`. For instance, an `angular.json` may contain projects configured as below.
```
Comment (isaacplmann) [inline docs/node/guides/configuration.md:309]: should be workspace.json not angular.json

File: docs/node/guides/configuration.md [hunk 2]
```diff
@@ -302,6 +304,33 @@ The following command will generate a new library: `nx g @nrwl/node:lib mylib`.
...
+
+```json
+{
+  "projects": {
+    "mylib": "libs/mylib"
+  }
+}
+```
+
+This tells Nx that all configuration for that project is found in the `libs/mylib/project.json` file. This file contains a combination of the project's configuration from both `angular.json` and `nx.json`.
```
Comment (isaacplmann) [inline docs/node/guides/configuration.md:319]: workspace.json

File: docs/react/guides/configuration.md [hunk 3]
```diff
@@ -290,6 +292,33 @@ The following command will generate a new library: `nx g @nrwl/react:lib mylib`.
 
 When the `version` of `workspace.json` is set to 2, `targets`, `generators` and `executor` properties are used instead of the version 1 properties `architect`, `schematics` and `builder`.
 
+## project.json
+
+Project configurations can also be independent files, referenced by `angular.json`. For instance, an `angular.json` may contain projects configured as below.
```
Comment (isaacplmann) [inline docs/react/guides/configuration.md:297]: workspace.json

File: docs/react/guides/configuration.md [hunk 4]
```diff
@@ -290,6 +292,33 @@ The following command will generate a new library: `nx g @nrwl/react:lib mylib`.
...
+
+```json
+{
+  "projects": {
+    "mylib": "libs/mylib"
+  }
+}
+```
+
+This tells Nx that all configuration for that project is found in the `libs/mylib/project.json` file. This file contains a combination of the project's configuration from both `angular.json` and `nx.json`.
```
Comment (isaacplmann) [inline docs/react/guides/configuration.md:307]: workspace.json

**Review by isaacplmann — CHANGES_REQUESTED** [review]
Change a few references to angular.json to workspace.json.

Otherwise looks good

File: docs/node/guides/configuration.md (hunk 1 above)
Comment (AgentEnder) [reply docs/node/guides/configuration.md]: Thanks for catching these, I missed them when copying the statements around between the 3 versions.

**Review by isaacplmann — APPROVED** [review]

**Pull request merged**